File: backend/app/schemas/statistics.py
from datetime import datetime

from pydantic import BaseModel


class PeriodStatsItem(BaseModel):
    period_label: str
    period_start: datetime
    activity_count: int
    total_distance: float
    total_moving_time: int
    avg_pace_sec_per_km: float | None
    avg_heartrate: float | None
    total_elevation_gain: float


class WeeklyStatsResponse(BaseModel):
    year: int
    month: int
    weeks: list[PeriodStatsItem]


class MonthlyStatsResponse(BaseModel):
    year: int
    months: list[PeriodStatsItem]


class YearlyStatsResponse(BaseModel):
    years: list[PeriodStatsItem]

File: frontend/src/api/statistics.ts
import apiClient from './client'

export interface PeriodStatsItem {
  period_label: string
  period_start: string
  activity_count: number
  total_distance: number
  total_moving_time: number
  avg_pace_sec_per_km: number | null
  avg_heartrate: number | null
  total_elevation_gain: number
}

export interface WeeklyStatsResponse {
  year: number
  month: number
  weeks: PeriodStatsItem[]
}

export interface MonthlyStatsResponse {
  year: number
  months: PeriodStatsItem[]
}

export interface YearlyStatsResponse {
  years: PeriodStatsItem[]
}

export const fetchWeeklyStats = (year: number, month: number) =>
  apiClient.get<WeeklyStatsResponse>('/statistics/weekly', { params: { year, month } })

export const fetchMonthlyStats = (year: number) =>
  apiClient.get<MonthlyStatsResponse>('/statistics/monthly', { params: { year } })

export const fetchYearlyStats = () => apiClient.get<YearlyStatsResponse>('/statistics/yearly')
